builtin/printf_osh.py

```diff
@@ -49,6 +49,53 @@
 _ = log
 
 
+def _ParsePrintfInteger(s, blame_loc):
+    # type: (str, loc_t) -> Tuple[bool, mops.BigInt]
+    """
+    Returns:
+      (True, value) when the string looks like an integer
+      (False, ...)  when it doesn't
+
+    Integer formats that are recognized:
+      0xAB    hex
+      042     octal
+      42      decimal
+    """
+    # Normalize 0X to 0x for the lexer
```

Comment:
Oh actually this is a bug in OSH!

```
$ bash -c 'echo $(( 0XAA ))'
170

$ osh -c 'echo $(( 0XAA ))'
  echo $(( 0XAA ))
           ^
[ -c flag ]:1: fatal: Invalid integer constant '0XAA'
```

Let's change the lexer instead - search for `ShNumber_Hex`